mono/mini/mini.c

```diff
@@ -3577,6 +3593,11 @@ mini_method_compile (MonoMethod *method, guint32 opts, MonoDomain *domain, JitFl
 		mono_cfg_dump_ir (cfg, "local_cprop");
 	}
 
+	MONO_TIME_TRACK (mono_jit_stats.jit_decompose_typechecks, mono_decompose_typechecks (cfg));
```

Comment:
Yes, we should do it as late as possible.  IIRC there're some issues around inserting new basic blocks after `mono_optimize_branches`, I'll look into that again.